server/py/services/api/db/sqldb/db.py

```diff
@@ -5384,6 +5384,90 @@ def _enrich_notification(_notification):
             for cooldown, notification in zip(cooldowns, notifications)
         ]
 
+    @staticmethod
+    def create_partitions(
+        session: Session,
+        table_name: str,
+        partitioning_information_list: list[tuple[str, str, str]],
+    ):
+        """
+        Creates partitions in the specified database table.
+
+        :param session: SQLAlchemy session for database connection.
+        :param table_name: Name of the table where partitions will be created.
+        :param partitioning_information_list: List of tuples, each containing:
+            - partition_name: The name for the partition.
+            - partition_value: The "LESS THAN" boundary value for the partition.
+            - partition_expression: The SQL partition expression.
+        """
+        alter_table_template = f"""
+            ALTER TABLE {table_name}
+            PARTITION BY RANGE ({partitioning_information_list[0][2]}) (
+                {", ".join([
+            f"PARTITION p{partition_name} VALUES LESS THAN ({partition_value})"
+            for partition_name, partition_value, _ in partitioning_information_list
+        ])}
+            );
+        """
+
+        # Execute the partitioning SQL
+        session.execute(text(alter_table_template))
+        session.commit()
+
+    @staticmethod
+    def drop_partitions(
+        session: Session,
+        table_name: str,
+        cutoff_partition_name: str,
+    ):
+        """
+        Execute the drop operation for partitions older than the cutoff partition name.
+
+        :param session: SQLAlchemy session.
+        :param table_name: The name of the table with partitions.
+        :param cutoff_partition_name: The cutoff partition name for dropping old partitions.
+        """
+        # Query partitions older than cutoff_partition_name
+        partitions_to_drop = session.execute(
+            text("""
+            SELECT GROUP_CONCAT(PARTITION_NAME) AS partitions
+            FROM information_schema.PARTITIONS
+            WHERE TABLE_NAME = :table_name
+              AND PARTITION_NAME < :cutoff_partition_name
+        """),
```

Comment:
@Yacouby since `information_schema.PARTITIONS` is a MySQL system table, you cannot modify its schema or add indexes to it directly. The query performance on information_schema tables can be slower when there are many partitions, but indexing or altering information_schema tables is not possible in MySQL.